vite.config.js

```diff
@@ -1,19 +1,67 @@
 import { sentryVitePlugin } from '@sentry/vite-plugin';
 import { fileURLToPath, URL } from 'url';
 import { defineConfig } from 'vite';
-import vitePluginFaviconsInject from 'vite-plugin-favicons-inject';
 import Vue from '@vitejs/plugin-vue';
-import basicSsl from '@vitejs/plugin-basic-ssl';
 import { nodePolyfills } from 'vite-plugin-node-polyfills';
+import { VitePWA } from 'vite-plugin-pwa';
+import mkcert from 'vite-plugin-mkcert';
 
 // https://vitejs.dev/config/
 export default defineConfig({
   plugins: [
     Vue({
       include: [/\.vue$/, /\.md$/],
     }),
-    vitePluginFaviconsInject('./src/assets/roar-icon.svg'),
-    ...(process.env.NODE_ENV === 'development' ? [basicSsl()] : []),
+    VitePWA({
+      manifest: {
+        // Modify manifest options here...
+        name: 'ROAR Dashboard',
+        short_name: 'ROAD',
+        start_url: '.',
+        display: 'standalone',
+        theme_color: '#ffffff',
+        background_color: '#ffffff',
+        // inject service worker automatically
+        strategies: 'injectManifest',
+        injectRegister: 'manual',
+        workbox: {
+          globPatterns: ['**/*.{js,css,html,ico,png,svg,mp3}'],
+        },
+        icons: [
+          {
+            src: 'pwa-64x64.png',
+            sizes: '64x64',
+            type: 'image/png',
+          },
+          {
+            src: 'pwa-192x192.png',
+            sizes: '192x192',
+            type: 'image/png',
+          },
+          {
+            src: 'pwa-512x512.png',
+            sizes: '512x512',
+            type: 'image/png',
+            purpose: 'any',
+          },
+          {
+            src: 'maskable-icon-512x512.png',
+            sizes: '512x512',
+            type: 'image/png',
+            purpose: 'maskable',
+          },
+        ],
+        // Add more manifest options as needed
+      },
+      /* enable sw on development */
+      devOptions: {
+        enabled: true,
+        type: 'module',
+        /* other options */
+      },
+      registerType: 'autoUpdate',
```

Comment:
Autoupdate is easy but can cause inconvenience in production. When auto update triggers it does not save the current workflow and refreshes the page which can cause issues. I personally like the user to manually refresh it:
https://vite-pwa-org.netlify.app/guide/prompt-for-update.html
A lot of apps seem to follow this like Shortcut etc